Kconfig.nrf

```diff
@@ -8,23 +8,6 @@ rsource "subsys/net/openthread/Kconfig.defconfig"
 
 menu "Nordic nRF Connect"
 
-#
-# Provide a new choice to override the mbedtls_external library completely
-# and not have to provide a "dummy" path for the implementation
-#
-choice MBEDTLS_IMPLEMENTATION
-	prompt "Select implementation"
-	default MBEDTLS_LIBRARY_NRF_SECURITY
-
-config MBEDTLS_LIBRARY_NRF_SECURITY
-	bool "Use mbed TLS library from Nordic provided security backend"
-	depends on NORDIC_SECURITY_BACKEND || NRF_SECURITY
-	help
-	  Use external, out-of-tree prebuilt mbedTLS library. For advanced
-	  users only.
-
-endchoice
```

Comment:
this is extending the upstream choice menu from here:
https://github.com/nrfconnect/sdk-zephyr/blob/df384c23a1a463623a6b0bb57ce1a726515a13ec/modules/mbedtls/Kconfig#L23

to allow users to use nRF Security.

To my knowledge this cannot be removed, but @frkv can confirm.

Reply:
This was moved to the Kconfig file in `/subsys`

Reply:
didn't notice that, thanks for the update.

Reply:
Yes. I saw this PR and approved it because the configs aren't removed, they are just moved

When I think about it there is one thing that may be important to verify and that is that the subsys n use is not taken in before Zephyr (which we are overriding)

Reply:
the commit message says:
> Try to clean up the top-level Kconfig menu.

but the reality is that this setting is still located in the same top-level nrf menu, just a bit further down now.
`sdk-nrf/main`:
![image](https://user-images.githubusercontent.com/9802583/188836874-ddafd7b2-6b91-4147-9900-800ba4592ebf.png)

PR #7996:
![image](https://user-images.githubusercontent.com/9802583/188837094-6458329c-dac1-4c76-94e3-274b729922c8.png)

so the top-level Kconfig menu is not really cleaned up.

Was this your intention with the PR ?